packages/framework/src/Component/Grid/QueryBuilderDataSourceFactory.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\Grid;
+
+use Doctrine\ORM\QueryBuilder;
+
+class QueryBuilderDataSourceFactory
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Grid\HintsHelper $hintsHelper
+     */
+    public function __construct(
+        protected readonly HintsHelper $hintsHelper,
+    ) {
+    }
+
+    /**
+     * @param \Doctrine\ORM\QueryBuilder $queryBuilder
+     * @param string $rowIdSourceColumnName
+     * @param array|null $hints
+     * @return \Shopsys\FrameworkBundle\Component\Grid\QueryBuilderDataSource
+     */
+    public function create(
+        QueryBuilder $queryBuilder,
+        string $rowIdSourceColumnName,
+        ?array $hints = null,
+    ): QueryBuilderDataSource {
+        return new QueryBuilderDataSource(
+            $queryBuilder,
+            $rowIdSourceColumnName,
+            $hints ?? $this->hintsHelper->getDefaultHints(),
```

Comment:
Oh, I see now the default hints are set here... I am for removing the default value from QueryBuilderDataSource constructor